service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java

```diff
@@ -1304,6 +1375,40 @@ private PaymentRequestDto formPayment(Long orderId, int sumToPay) {
         return paymentRequestDto;
     }
 
+    private int formCertificatesToBeSavedAndCalculateOrderSumClient(OrderFondyClientDto dto,
+        Set<Certificate> orderCertificates,
+        Order order, int sumToPay) {
+        if (sumToPay != 0 && dto.getCertificates() != null) {
+            boolean tooManyCertificates = false;
+            for (String temp : dto.getCertificates()) {
+                if (tooManyCertificates) {
+                    throw new TooManyCertificatesEntered(TOO_MANY_CERTIFICATES);
+                }
+                Certificate certificate = certificateRepository.findById(temp).orElseThrow(
+                    () -> new CertificateNotFoundException(CERTIFICATE_NOT_FOUND_BY_CODE + temp));
+                validateCertificate(certificate);
+                certificate.setOrder(order);
+                orderCertificates.add(certificate);
+                sumToPay -= certificate.getPoints();
+                certificate.setCertificateStatus(CertificateStatus.USED);
+                certificate.setDateOfUse(LocalDate.now());
+                if (dontSendLinkToFondyIfClient(sumToPay, certificate, dto)) {
+                    sumToPay = 0;
+                    tooManyCertificates = true;
+                }
+            }
+        }
+        return sumToPay;
+    }
+
+    private boolean dontSendLinkToFondyIfClient(int sumToPay, Certificate certificate, OrderFondyClientDto dto) {
```

Comment:
this method does 2 things: answers the question  as well as modifies object state. This is bad practice. Method should do either modification or answer.